lib/lure/version.go

```diff
@@ -0,0 +1,105 @@
+package lure
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	versionRegex *regexp.Regexp
+)
+
+const (
+	versionRegexStr = `(\d+)(?:[.-](\d+))?(?:[.-](\d+))?(?:[.-](\d+))?[.-]?(.*)`
+)
+
+// Version represents a version with four numeric segments and an optional qualifier (e.g 1.2.3-RELEASE)

```

Comment:
three segments?

Reply:
J'ai laissé 4 en basant sur https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN401 qui en faisait mention. Dans un contexte plus "large", c'est assez commun, donc ca me paraissait raisonnable d'arrêter à 4.

Reply:
Your comment talk about 4 numerical segment than your example has 3

Reply:
Oh right, it never crossed my mind. At most four, but I should have added more examples to clear out the confusion)